index.html

```diff
@@ -1310,11 +1309,8 @@ <h2>Public Keys</h2>
 
       <p>
 If a public key does not exist in the <a>DID document</a>, it MUST be assumed
-the key was revoked or is invalid. The <a>DID document</a> MAY contain revoked
-keys. A <a>DID document</a> containing a revoked key MUST also contain or refer
-to the revocation information for the key (for example, a revocation list). Each
-<a>DID method</a> specification is expected to detail how revocation is
-performed and tracked.
+the key was revoked or is invalid. The <a>DID document</a> MUST NOT contain revoked
```

Comment:
There is not consensus on this second sentence that a DID Document MUST NOT contain revoked keys. There are good use cases for retaining revoked keys if the Controller wishes to communicate that certain keys were once valid but are no longer. The first sentence is fine, IMO. If it isn't in the document, it was revoked or never valid. The second one is overstating the consensus.

Reply:
The objective of this PR is to make the concept of revocation clearer... can you point me to any did method that uses revokes keys in a did document today? I suggest we start by agreeing that a did document will not contain revoked keys, and then if someone  (you?) wants to create a PR to define revoked keys, we address that PR separately...  I initially tried to do that here, and then updated this PR based on feedback from  @msporny @csuwildcat  and others....

Reply:
I can't point you to such methods, but I can point you to the comment thread that seemed to achieve consensus on this matter: https://github.com/w3c/did-core/issues/14#issuecomment-587565105 

The use case that is driving this comes from @ken-ebert: there is a desire to be able to evaluate older credentials based on whether or not a key was valid when the credential was created, even through that key is no longer appropriate for new credentials. However, typing that out, I would question whether or not you can know when a credential was created in order to apply the previously legitimate (and not known to be compromised) key. @ChristopherA has also expressed a desire to be able to state unequivocally that a past key is known to be compromised. This is the inverse of Ken's request, but has a similar construct: in some cases it may be appropriate to communicate some information about specific keys that have been revoked for specific reasons.

That said, my comment is less about the point of value of such annotations than about what appeared to have consensus previously. The consensus was NOT to restrict DID Documents from containing revoked keys, but rather to establish that a key that is not in the DID Document should be considered either revoked or never valid, leaving open the option for others to propose language that would clarify exactly how such a revoked key might be represented.

Reply:
I would welcome a PR from @ken-ebert or @ChristopherA defining revocation, but in its absence I think its a terrible idea to allow people to think that a DID Document can contain revoked keys... we should be crystal clear on matters that have such drastic security implications.

Reply:
I understand your position. But that was not the consensus for which a PR was requested.

Please restrict the PR to the consensus. If you would like to add your own opinion, it would be welcome in another issue where it can be debated appropriately.

Reply:
@jandrieu I believe you are misstating "consensus" in this case (as it applies to others, I concede that we clearly disagree :)

Of course, if others object to this PR text, I will concede that I have misinterpreted their preferences as well :)

The goal of this PR is to clarify the concept of "revoked keys"... removing the "MUST NOT contain revoked keys" does not achieve that goal.

Your request in essence, is "leaving the door open to interpretation" for one of the most critical aspects of PKI... with serious security implications.

It is my position that we (except for you) have reached consensus on this matter, and that you, or others who wish to define support for something that everyone else (@msporny @selfissued @csuwildcat @dlongley @msporny , etc...) does not want, should open a PR to do so.

If we are not going to explicitly forbid the inclusion of revoked keys, we would need to define them, and since you have already agreed that nobody is using them... how can we do that?

Reply:
Regarding consensus, I'm just quoting Manu's assertion that there seemed to be some consensus, so "let's write it up".

I don't have a problem if, in fact, Manu was over-eager on calling for consensus. You and I are apparently not in consensus, so we need to find a way toward that.

You say:
> If we are not going to explicitly forbid the inclusion of revoked keys, we would need to define them

Umm... no. We don't. The language of revoked keys is already there. In fact, your addition of a normative requirement that revoked keys MUST NOT be in the DID Document would require more definition than staying silent on the matter. The fewer constraints the less you are required to define.

My points are twofold. 

First, there was perceived consensus around my own comment on issue 14, cited above. This PR doesn't represent that consensus. That's fine, it seems like there was no consensus so we shouldn't have jumped to a PR yet. Which also means this PR isn't ready.

Second, forbidding revoked keys anywhere is unnecessary overkill at this point. There are identified use cases that want some way to put "old" keys in the DID Document, but there are, as yet, no PRs proposing exactly how that would be done. Those PRs are going to be necessary to enable some form of this capability and it is appropriate to wait until those are written to debate it further so we have something concrete to discuss. However, if you restrict revoked keys now, then you are falsely asserting a consensus that doesn't exist, which will make it harder for those PRs to have an honest hearing.

Both of these options currently lack consensus.

Reply:
`@ChristopherA has also expressed a desire to be able to state unequivocally that a past key is known to be compromised`

I'm not keen on explicitly stating that a key has been compromised, but I would augment that this PR should say: 

```
If a public key does not exist in the most recent version of the DID document, 
it MUST be assumed  the key was revoked or is invalid. 
```

 There is a need to query older versions of the DID document for validation for a time period that the VC was signed.

Reply:
These comments should be made on the issue, so everyone can see them.